pkg/artifactory/resource_xray_policy_test.go

```diff
@@ -0,0 +1,379 @@
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+			{
+				Config: testAccXrayPolicy_allActions(policyName, updatedDesc, updatedRuleName, updatedMail),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", policyName),
+					resource.TestCheckResourceAttr(resourceName, "description", updatedDesc),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.name", updatedRuleName),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.priority", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.criteria.0.min_severity", "High"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.actions.0.fail_build", "true"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.actions.0.mails.0", updatedMail),
+				),
+			},
+		},
+	})
+}
+
+func TestAccPolicy_licenseCriteria(t *testing.T) {
+	policyName   := "terraform-test-policy"
+	policyDesc   := "policy created by xray acceptance tests"
+	ruleName     := "test-security-rule"
+	allowedLicense := "BSD-4-Clause"
+	bannedLicense1 := "0BSD"
+	bannedLicense2 := "diffmark"
+	resourceName := "xray_policy.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckPolicyDestroy,
+		Providers:    testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccXrayPolicy_license(policyName, policyDesc, ruleName, allowedLicense),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", policyName),
+					resource.TestCheckResourceAttr(resourceName, "description", policyDesc),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.name", ruleName),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.priority", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.criteria.0.allowed_licenses.0", allowedLicense),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+			{
+				Config: testAccXrayPolicy_licenseBanned(policyName, policyDesc, ruleName, bannedLicense1, bannedLicense2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", policyName),
+					resource.TestCheckResourceAttr(resourceName, "description", policyDesc),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.name", ruleName),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.priority", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.criteria.0.banned_licenses.0", bannedLicense1),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.criteria.0.banned_licenses.1", bannedLicense2),
+				),
+			},
+		},
+	})
+}
+
+// This should be uncommented when someone figures out how to deal with this (see comment in expandActions in the provider)
```

Comment:
I don't see what the issue is

Reply:
I go into a bit more detail [here](https://github.com/jfrog/terraform-provider-artifactory/pull/36/files#diff-a42322c60382dc085e7f6a3cbbc6d4f7fe8e7882fe21e490537e0b618268e014R308), but tldr when `block_download` is optional/missing in the schema, you see a perpetual diff in the state even when nothing has changed. That isn't great from an end-user perspective, so the alternative that seems to work is making `block_download` required (even if its contents are just set to false).